### ISSUE TITLE
fails to print Derivative for latex

### ISSUE THREAD

**Issue by Alex031544** [sympy/sympy#16925]
Hello,

I create a symbolic expression and want do print it's derivation with latex.
I expect `\frac{d}{d t} q_{1}`, but I receive `\frac{d}{d t} \operatorname{q_{1}}{\left(t \right)}`

Look at this Python3 example:
```py
from sympy import *
from sympy import init_printing
init_printing(use_latex=True)

from sympy.core.function import Function

t = Symbol('t')
q1 = Function('q1')(t)
q1d = diff(q1,t)

print("q1: %s"%q1)
print("q1d: %s"%q1d)
print("latex(q1d): %s"%latex(q1d))
```
results in
```
q1: q1(t)
q1d: Derivative(q1(t), t)
latex(q1d): \frac{d}{d t} \operatorname{q_{1}}{\left(t \right)}
```

How can this be solved?

**Comment by oscarbenjamin** [comment]
Not sure how to solve it but it seems to be an issue with Functions that have more than a single letter name:
```
In [15]: latex(Function('f'))                                                                                                     
Out[15]: 'f'

In [16]: latex(Function('fa'))                                                                                                    
Out[16]: '\\operatorname{fa}'
```
That makes sense because if a function name has two letters then it should print upright. It doesn't make sense for subscripts though:
```
In [17]: latex(Function('f1'))                                                                                                    
Out[17]: '\\operatorname{f_{1}}'

In [18]: latex(Symbol('f1'))                                                                                                      
Out[18]: 'f_{1}'
```

**Comment by Alex031544** [comment]
on Symbols this does not happen:
```py
print(latex(Symbol('fa'))) 
```
prints
```sh
fa
```

**Comment by Alex031544** [comment]
in *sympy/printing/latex.py* I found:

```py
    def _print_Function(self, expr, exp=None):
        r'''
        Render functions to LaTeX, handling functions that LaTeX knows about
        e.g., sin, cos, ... by using the proper LaTeX command (\sin, \cos, ...).
        For single-letter function names, render them as regular LaTeX math
        symbols. For multi-letter function names that LaTeX does not know
        about, (e.g., Li, sech) use \operatorname{} so that the function name
        is rendered in Roman font and LaTeX handles spacing properly.

        expr is the expression involving the function
        exp is an exponent
        '''
```
This looks a bit strange - but it seems to be intended

**Comment by oscarbenjamin** [comment]
I think that this is a valid issue. The problem is here:
https://github.com/sympy/sympy/blob/da4929882a0492492549f8dd5a67f262df1ff142/sympy/printing/latex.py#L749-L752

This doesn't check to see whether the function is more than one letter because of a subscript (e.g. `f_{1}` rather than a multi-letter function name like `cos`.

**Comment by Alex031544** [comment]
Ah I understand - you mean it should be differentiated between `f1` and `fa` because:

- `f1` becomes `f_{1}`
- `fa` becomes `\operatorname{fa}`

**Comment by Alex031544** [comment]
Personally I would rather prefer the dot notation for time related one letter functions: `\dot{f}`

**Comment by oscargus** [comment]
The problem here is to know what a time related function is. This functionality exists in the physics module, but I do not know from the top of my head if it can be inherited or the exact requirements for it to be printed with a dot above.

I think that the one letter printing thing is solvable though.

**Comment by Alex031544** [comment]
What about changing:
```py
    def _hprint_Function(self, func):
        r'''
        Logic to decide how to render a function to latex
          - if it is a recognized latex name, use the appropriate latex command
          - if it is a single letter, just use that letter
          - if it is a longer name, then put \operatorname{} around it and be
            mindful of undercores in the name
        '''
        func = self._deal_with_super_sub(func)
        if func in accepted_latex_functions:
            name = r"\%s" % func
        elif len(func) == 1 or func.startswith('\\'):
            name = func
        else:
            name = r"\operatorname{%s}" % func
        return name
```
to:
```py
    def _hprint_Function(self, func):
        r'''
        Logic to decide how to render a function to latex
          - if it is a recognized latex name, use the appropriate latex command
          - if it is a single letter, just use that letter
          - if it is a longer name, then put \operatorname{} around it and be
            mindful of undercores in the name
        '''
        func = self._deal_with_super_sub(func)
        if func in accepted_latex_functions:
            name = r"\%s" % func
        elif len(func) == 1 or func.startswith('\\') or func.find("_") == 1 or func.find("^") == 1:
            name = func
        else:
            name = r"\operatorname{%s}" % func
        return name
```
Because `func = self._deal_with_super_sub(func)` already separates the super- and subscript from the name.  So it might be useful to detect if they exist. See the following example:
```py
funs = []
funs.append("str")
funs.append("str_1")
funs.append("str^1")
funs.append("str_1^2")
funs.append("str^1_2")
funs.append("s")
funs.append("s_1")
funs.append("s^1")
funs.append("s_1^2")
funs.append("s^1_2")

for fun in funs:
    print("%s: len=%d; _=%d; ^=%d"%(fun,len(fun),fun.find("_"),fun.find("^")))
```
which results to:
```
str: len=3; _=-1; ^=-1
str_1: len=5; _=3; ^=-1
str^1: len=5; _=-1; ^=3
str_1^2: len=7; _=3; ^=5
str^1_2: len=7; _=5; ^=3
s: len=1; _=-1; ^=-1
s_1: len=3; _=1; ^=-1
s^1: len=3; _=-1; ^=1
s_1^2: len=5; _=1; ^=3
s^1_2: len=5; _=3; ^=1
```

**Comment by oscarbenjamin** [comment]
Yeah that could work. Can you open a PR?

**Comment by Alex031544** [comment]
Yes

**Comment by Alex031544** [comment]
done https://github.com/sympy/sympy/pull/16968

**Comment by asmeurer** [comment]
Don't close the issue until the PR is merged. If you put "fixes #16968" in the PR description, it will automatically close this issue when it is merged.

**Comment by czgdp1807** [comment]
Update:
The issue still persists on `master`.
